x/posts/internal/types/keys.go

```diff
@@ -28,6 +28,7 @@ const (
 
 var (
 	SubspaceRegEx = regexp.MustCompile("^[a-fA-F0-9]{64}$")
+	HashtagRegEx  = regexp.MustCompile(`(?:|^)#[A-Za-z0-9]+(?:|$)`)
```

Comment:
Still, If we look at Twitter, this is what happens: 

| String | Hashtags found |
| :------ | :------------------| 
| `#first#second` | `[]` |
| `#first #second` | `[first, second]`
| `#first_#second` | `[]` |

So it's considered an hashtag only if it's at the beginning of the message or is after a white space character (space, newline, etc). 

That being said, I think the RegEx should be changed to be

```suggestion
	HashtagRegEx  = regexp.MustCompile(`(\s|^)#[A-Za-z0-9]+(\s|$)`)
```

Reply:
Trying this regex and it seems not working

Reply:
Probably the leading `/` and trailing `/g` must be removed: 

```
\s([#][a-zA-Z_]+)
```

Reply:
Since even the latest regEx doesn't match all the cases you mentioned before, I surfed on the web and combined a solution that match all the cases (including numeric hashtags).
Let me know what do you think about it. I've also added more testcases to make sure that it's correct!